web_src/js/features/repo-home.js

```diff
@@ -152,12 +155,13 @@ export function initRepoTopicBar() {
   });
 
   $.fn.form.settings.rules.validateTopic = function (_values, regExp) {
-    const $topics = $topicDropdown.children('a.ui.label');
-    const status = !$topics.length || $topics.last()[0].getAttribute('data-value').match(regExp);
-    if (!status) {
-      $topics.last().removeClass('green').addClass('red');
+    const topics = topicDropdown.querySelectorAll(':scope > a.ui.label');
+    const status = !topics.length || (topics[topics.length - 1]).getAttribute('data-value').match(regExp);
+    if (!status && topics[topics.length - 1]) {
+      (topics[topics.length - 1]).classList.remove('green');
+      (topics[topics.length - 1]).classList.add('red');
```

Comment:
TBH I am not a fan of such refactoring style. As a refactoring PR, if it could be done with simple search&replace, then it could keep the old logic, and some tests could be skipped.

However, since it has been rewritten a lot and needs to be fully tested, I believe it's better to rewrite the code to a proper style.


```suggestion
    const topics = topicDropdown.querySelectorAll(':scope > a.ui.label');
    const lastTopic = topics[topics.length - 1];
    const isLastTopicInvalid = lastTopic?.getAttribute('data-value').match(regExp);
    if (isLastTopicInvalid) {
      lastTopic.classList.remove('green');
      lastTopic.classList.add('red');
```